src/shared/version_op.c

```diff
@@ -654,6 +665,24 @@ os_info *get_unix_version()
             }
             regfree(&regexCompiled);
             fclose(version_release);
+        // Alpine
+        } else if (version_release = fopen("/etc/alpine-release","r"), version_release){
+            info->os_name = strdup("Alpine Linux");
+            info->os_platform = strdup("alpine");
+            static const char *pattern = "([0-9]+\\.)?([0-9]+\\.)?([0-9]+)";
```

Comment:
Why not something like this to avoid changing the match size?             
`static const char *pattern = "([0-9]+\\.[0-9]+\\.[0-9]+)";`

Reply:
Because regex you propose "([0-9]+\\.[0-9]+\\.[0-9]+)" only match if 3 version number blocks "major.minor.patch" are present.
And regex "([0-9]+\\.)?([0-9]+\\.)?([0-9]+)" match 3 possibilities: "major", "major.minor" and "major.minor.patch".